compiler-core/src/ast/typed.rs

```diff
@@ -97,11 +97,27 @@ pub enum TypedExpr {
     },
 
     ModuleSelect {
+        /// The location of the selected value coming after the `.` (including
+        /// it):
+        ///
+        /// ```gleam
+        ///    wibble.wobble
+        /// //       ^^^^^^^ location
+        /// ```
+        ///
         location: SrcSpan,
         type_: Arc<Type>,
         label: EcoString,
         module_name: EcoString,
         module_alias: EcoString,
+        /// The location of the module before the `.`:
+        ///
+        /// ```gleam
+        ///    wibble.wobble
+        /// // ^^^^^^ module_location
+        /// ```
+        ///
+        module_location: SrcSpan,
```

Comment:
The second int is redundant, it is the same as the first int in location.

Perhaps we want to have location be the whole thing and then have a `field_start` or something which is the location of the first character after the `.`? Would be nice to have the warning and errors from from after the dot rather than on it too so you don't have to press right after jumping to the error to fix it.

Reply:
I would love for the `location` to cover the entire field, I've always found it a bit odd that the module selection location doesn't cover the whole thing, and it means we always have to special case those in code actions that need to work on their entire span. I'll try and fix it!

Reply:
Thanks Jak!

Reply:
Done! <https://github.com/gleam-lang/gleam/pull/4237/commits/fafcfb8e2c309835ebce96ad5809e6e63110ba76>